File: components_page/components/tooltip/placement.py
import dash_bootstrap_components as dbc
import dash_html_components as html


def make_button(placement):
    return dbc.Button(
        f"Tooltip on {placement}",
        id=f"tooltip-target-{placement}",
        className="mx-2",
    )


def make_tooltip(placement):
    return dbc.Tooltip(
        f"Tooltip on {placement}",
        target=f"tooltip-target-{placement}",
        placement=placement,
    )


tooltips = html.Div(
    [make_button(p) for p in ["top", "left", "right", "bottom"]]
    + [make_tooltip(p) for p in ["top", "left", "right", "bottom"]]
)

File: components_page/components/tooltip/placement.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

make_button <- function(placement) {
  return(
    dbcButton(
      sprintf("Tooltip on %s", placement),
      id = sprintf("tooltip-target-%s", placement),
      n_clicks = 0,
      className = "mx-2"
    )
  )
}


make_tooltip <- function(placement) {
  return(
    dbcTooltip(
      sprintf("Tooltip on %s", placement),
      target = sprintf("tooltip-target-%s", placement),
      placement = placement
    )
  )
}


tooltips <- htmlDiv(
  list(
    make_button("top"), make_button("left"),
    make_button("right"), make_button("bottom"),
    make_tooltip("top"), make_tooltip("left"),
    make_tooltip("right"), make_tooltip("bottom")
  )
)
